test/utils.h

```diff
@@ -403,7 +403,7 @@ inline bool maybeClearAllocator(int64_t max_bytes = ((int64_t)1 << 32)) {
     auto device_stats = allocator->getDeviceStats(0);
     // allocated_bytes[] holds multiple statistics but the first is sum across
     // both small and large blocks
-    if (device_stats.reserved_bytes[0].current > max_bytes) {
+    if (device_stats.reserved_bytes[0].current > uint64_t(max_bytes)) {
```

Comment:
I'm getting this error when trying to build in the pnjl-latest container (gcc 11.4.0):
> In file included from /opt/pytorch/nvfuser/csrc/kernel_db/test/test_nvfuser_kernel_db_write.cpp:15:                                   
/opt/pytorch/nvfuser/test/utils.h: In function ‘bool nvfuser::maybeClearAllocator(int64_t)’:                                          
/opt/pytorch/nvfuser/test/utils.h:406:48: error: comparison of integer expressions of different signedness: ‘int64_t’ {aka ‘long int’}
 and ‘uint64_t’ {aka ‘long unsigned int’} [-Werror=sign-compare]                                                                                
  406 |     if (device_stats.reserved_bytes[0].current > uint64_t(max_bytes)) {                                                       
cc1plus: all warnings being treated as errors                                                                                         
[269/378] Building CXX object CMakeFiles/nvfuser_tests.dir/test/utils.cpp.o
FAILED: CMakeFiles/nvfuser_tests.dir/test/utils.cpp.o

Reply:
This is surprising as it seems like the fix is correct since `device_stats.reserved_bytes[0].current` should be uint64_t: https://github.com/pytorch/pytorch/blob/main/c10/cuda/CUDACachingAllocator.h#L47

Reply:
This looks strange.... :confused:

Reply:
Err, it's weird. I think @jjsjann123 was trying to patch a similar build failure that we had yesterday.

Reply:
This is changed recently:
https://github.com/pytorch/pytorch/commit/9c7391ea369951f419ed6fa74e0de146185bb301#diff-b9a0773c48ce53900355ab72505056fae68c7117993028c826c9cc1e3f3c502d

Reply:
They changed it, then reverted it.